src/daft-table/src/ops/joins/hash_join.rs

```diff
@@ -43,32 +85,118 @@ pub(super) fn hash_inner_join(left: &Table, right: &Table) -> DaftResult<(Series
         ));
     }
 
-    // TODO(Clark): Build the probe table on the smaller table, rather than always building it on the left table.
+    let probe_left = match (how, left.len() <= right.len()) {
+        (JoinType::Left, _) => false,
+        (JoinType::Right, _) => true,
+        (_, left_side_smaller) => left_side_smaller,
+    };
```

Comment:
Nice, good consolidation of logic _and_ we get that "build probe table on smaller side" optimization for free! 👏